### PR TITLE
Changed ViewState and XFrameOption to return example alerts for the docs

### PR DESCRIPTION
**Pull request by psiinon**
Part of https://github.com/zaproxy/zaproxy/issues/6119

Associated script changes will follow v shortly...

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanRule.java [hunk 1]
```diff
@@ -188,4 +189,23 @@ private String getMetaXFOEvidence(Source source) {
...
     }
+
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<Alert>();
+        for (VulnType vulnType : VulnType.values()) {
+            alerts.add(
+                    newAlert()
+                            .setName(getAlertElement(vulnType, "name"))
+                            .setRisk(Alert.RISK_MEDIUM)
+                            .setConfidence(Alert.CONFIDENCE_MEDIUM)
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanRule.java:200]: Do you care that this doesn't account for:
https://github.com/zaproxy/zap-extensions/blob/7261748f7dfc5c4762d272eab880a439707b9778/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanRule.java#L116-L121

Is "Other Info" included in the docs?

Comment (kingthorin) [reply addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanRule.java:200]: Oops I guess this should only be associated with line 199.

Comment (psiinon) [reply addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanRule.java:200]: No - they are just example alerts. I dont think we can document every possible variation.

**Comment by thc202** [comment]
Thank you!

**Pull request merged**